src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/java/codeaction/JavaCodeActionContext.java

```diff
@@ -1,138 +1,138 @@
-	}
-
-	@Override
-	public PsiElement getCoveredNode() {
-		return getASTRoot().findElementAt(getSelectionOffset());
-	}
-
-	public JavaCodeActionContext getSource() {
-		return source;
-	}
-
-	public CodeAction convertToCodeAction(ChangeCorrectionProposal proposal, Diagnostic... diagnostics) {
-		String name = proposal.getName();
-		WorkspaceEdit edit = ChangeUtil.convertToWorkspaceEdit(proposal.getChange(), getUri(), getUtils(),
-				params.isResourceOperationSupported());
-		if (!ChangeUtil.hasChanges(edit)) {
-			return null;
-		}
-		ExtendedCodeAction codeAction = new ExtendedCodeAction(name);
-		codeAction.setRelevance(proposal.getRelevance());
-		codeAction.setKind(proposal.getKind());
-		codeAction.setEdit(edit);
-		codeAction.setDiagnostics(Arrays.asList(diagnostics));
-		return codeAction;
-	}
-
-	public WorkspaceEdit convertToWorkspaceEdit(ChangeCorrectionProposal proposal) {
-		WorkspaceEdit edit = ChangeUtil.convertToWorkspaceEdit(proposal.getChange(), getUri(), getUtils(),
-				params.isResourceOperationSupported());
-		if (!ChangeUtil.hasChanges(edit)) {
-			return null;
-		}
-		return edit;
-	}
-
-}
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.intellij.lsp4mp4ij.psi.core.java.codeaction;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.util.PsiUtil;
+import com.redhat.devtools.intellij.lsp4mp4ij.psi.core.java.AbstractJavaContext;
+import com.redhat.devtools.intellij.lsp4mp4ij.psi.core.java.corrections.proposal.ChangeCorrectionProposal;
+import com.redhat.devtools.intellij.lsp4mp4ij.psi.core.utils.IPsiUtils;
+import com.redhat.devtools.intellij.lsp4mp4ij.psi.internal.core.java.ChangeUtil;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.eclipse.lsp4mp.commons.MicroProfileJavaCodeActionParams;
+
+import java.util.Arrays;
+
+/**
+ * Java codeAction context for a given compilation unit.
+ *
+ * @author Angelo ZERR
+ *
+ */
+public class JavaCodeActionContext extends AbstractJavaContext implements IInvocationContext {
+
+	private final int selectionOffset;
+	private final int selectionLength;
+
+	private final MicroProfileJavaCodeActionParams params;
+	protected JavaCodeActionContext source;

```

Comment:
@angelozerr afaics source was changed from `private` to `protected`. That's the only change in this class. The rest are just changes in the formatting. Correct?